tests/int/test_headers.py

```diff
@@ -0,0 +1,174 @@
+import pytest
+from pytest_check import check
+import requests
+import os
+import asyncio
+from api_test_utils.oauth_helper import OauthHelper
+from api_test_utils.apigee_api_apps import ApigeeApiDeveloperApps
+from api_test_utils.apigee_api_products import ApigeeApiProducts
+from time import time
+
+
+ENVIRONMENT = os.environ['ENVIRONMENT']
+SERVICE_BASE_PATH = os.environ['SERVICE_BASE_PATH']
+SERVICE_NAME = os.environ['FULLY_QUALIFIED_SERVICE_NAME']
+BASE_URL = f'https://{ENVIRONMENT}.api.service.nhs.uk/{SERVICE_BASE_PATH}'
+
+ID_TOKEN_TESTING_PRIVATE_KEY_PATH = os.environ["ID_TOKEN_TESTING_PRIVATE_KEY_ABSOLUTE_PATH"]
+
+HEADER_AUTHORIZATION = 'Authorization'
+HEADER_ECHO = 'echo'
+HEADER_BASE_URL = 'x-ers-network-baseurl'
+HEADER_USER_ID = 'x-ers-user-id'
+HEADER_X_REQUEST_ID = 'x-request-id'
+HEADER_ASID = 'xapi_asid'
+
+ORIGINAL_HEADERS = {
+    "CORRELATION_ID" : "x-correlation-id",
+    "BUSINESS_FUNCTION" : "nhsd-ers-business-function",
+    "ODS_CODE" : "nhsd-ers-ods-code",
+    "FILENAME" : "nhsd-ers-file-name",
+    "COMM_RULE_ORG" : "nhsd-ers-comm-rule-org",
+    "REFERRAL_ID" : "nhsd-ers-referral-id"
+}
+
+RENAMED_HEADERS = {
+    "CORRELATION_ID" : "nhsd-correlation-id",
+    "BUSINESS_FUNCTION" : "x-ers-business-function",
+    "ODS_CODE" : "x-ers-ods-code",
+    "FILENAME" : "x-ers-xapi-meta-file_name",
+    "COMM_RULE_ORG" : "x-ers-xapi-comm-rule-org",
+    "REFERRAL_ID" : "x-ers-xapi-meta-intended_ubrn"
+}
+
+@pytest.mark.smoke_test
+class TestHeaders:
+
+    @pytest.fixture
+    async def product(self):
+        # Setup
+        product = ApigeeApiProducts()
+        await product.create_new_product()
+
+        print('GOT PRODUCT NAME: '  + product.name)
+
+        # Update products allowed paths
+        await product.update_proxies([SERVICE_NAME, 'identity-service-' + ENVIRONMENT])
+
+        await product.update_scopes(
+            scopes=["urn:nhsd:apim:user-nhs-id:aal3:e-referrals-service-api"]
+        )
+        yield product
+
+        #Teardown
+        print('Cleanup product')
+        await product.destroy_product()
+
+
+    @pytest.fixture
+    async def app(self, asid = '1234567890'):
+         # Setup
+        app = ApigeeApiDeveloperApps()
+        await app.create_new_app()
+
+        # Set default JWT Testing resource url and ASID
+        await app.set_custom_attributes({
+                'asid': asid,
+                'jwks-resource-url': 'https://raw.githubusercontent.com/NHSDigital/'
+                                     f'identity-service-jwks/main/jwks/{ENVIRONMENT}/'
+                                     '9baed6f4-1361-4a8e-8531-1f8426e3aba8.json'
+        })
+        yield app
+
+        #Teardown
+        print('Cleanup app')
+        await app.destroy_app()
+
+
+    async def get_access_code(self, appToUse, user_id):
```

Comment:
Very minor and personal preference but I think app instead of appToUse would make this simpler. There's only 1 app in question on this function and the mixture of camel and snake case is a bit jarring.

Reply:
The reason I changed it to appToUse was that my understanding (first time using python and pytest) is that if I named it app it would be auto-injected with a new app from the fixture whereas I wanted to pass one in?

Reply:
Ahh yeah I get you. Maybe test_app then or at least app_to_use. Or if you end up merging the app and product fixtures to test_app_and_product or something then it would be free.

Reply:
As it wasn't being used to dynamically inject a value I could have changed it to app. Since then I have now refactored it so that the access code itself is a fixture bringing in apps and products and the test is passing a user id as a paremter.